cylc/flow/scripts/cat_log.py

```diff
@@ -586,7 +586,15 @@ def _main(
             workflow_id, point, task, submit_num
         )
 
-        job_log_present = (Path(local_log_dir) / "job.out").exists()
+        job_log_present = all(
+            (Path(local_log_dir) / file).exists()
+            for file in [
+                # the files which are used to indicate that job log retrieval
+                # has completed
+                'job.out',
+                *platform['retrieve job log expected files'],
+            ]
+        )
```

Comment:
@ChrisPaulBennett 

With this PR, the condition we use to determine whether job log retrieval has completed can also take into account these configured files. The `cylc cat-log` logic should also respect this.